framework/test/unit/modules/validators/api.spec.ts

```diff
@@ -0,0 +1,426 @@
+
+		it(`should not be able to set generator key for validator if address does not exist`, async () => {
+			apiContext = new APIContext({ stateStore, eventQueue: new EventQueue() });
+
+			expect(
+				await validatorsModule.api.setValidatorGeneratorKey(apiContext, address, generatorKey),
+			).toBe(false);
+		});
+	});
+
+	describe('isKeyRegistered', () => {
+		it(`should return true if bls key is already registered`, async () => {
+			await blsKeysSubStore.set(blsKey, address);
+			apiContext = new APIContext({ stateStore, eventQueue: new EventQueue() });
+
+			expect(await validatorsModule.api.isKeyRegistered(apiContext, blsKey)).toBe(true);
+		});
+
+		it(`should return false if bls key is not registered`, async () => {
+			apiContext = new APIContext({ stateStore, eventQueue: new EventQueue() });
+
+			expect(await validatorsModule.api.isKeyRegistered(apiContext, blsKey)).toBe(false);
+		});
+	});
+
+	describe('getSlotNumber', () => {
+		it(`should return if timestamp is greater than genesis block timestamp`, async () => {
+			await genesisDataSubStore.setWithSchema(
+				emptyKey,
+				{ timestamp: genesisTimestamp },
+				genesisDataSchema,
+			);
+			apiContext = new APIContext({ stateStore, eventQueue: new EventQueue() });
+
+			expect(
+				await validatorsModule.api.getSlotNumber(apiContext, genesisTimestamp + 100),
+			).toBeNumber();
+		});
+
+		it(`should throw if timestamp is less than genesis block timestamp`, async () => {
+			await genesisDataSubStore.setWithSchema(
+				emptyKey,
+				{ timestamp: genesisTimestamp },
+				genesisDataSchema,
+			);
+			apiContext = new APIContext({ stateStore, eventQueue: new EventQueue() });
+
+			await expect(
+				validatorsModule.api.getSlotNumber(apiContext, genesisTimestamp - 100),
+			).rejects.toThrow('Invalid timestamp');
+		});
+	});
+
+	describe('getGeneratorAtTimestamp', () => {
+		it(`should return if timestamp is greater than genesis block timestamp`, async () => {
+			await genesisDataSubStore.setWithSchema(
+				emptyKey,
+				{ timestamp: genesisTimestamp },
+				genesisDataSchema,
+			);
+			await generatorListSubStore.setWithSchema(
+				emptyKey,
+				{ addresses: generatorList.map(addr => Buffer.from(addr, 'hex')) },
+				generatorListSchema,
+			);
+			apiContext = new APIContext({ stateStore, eventQueue: new EventQueue() });
+
+			expect(
+				await validatorsModule.api.getGeneratorAtTimestamp(apiContext, genesisTimestamp + 100),
+			).toBeInstanceOf(Buffer);
+		});
+
+		it(`should throw if timestamp is less than genesis block timestamp`, async () => {
+			await genesisDataSubStore.setWithSchema(
+				emptyKey,
+				{ timestamp: genesisTimestamp },
+				genesisDataSchema,
+			);
+			apiContext = new APIContext({ stateStore, eventQueue: new EventQueue() });
+
+			await expect(
+				validatorsModule.api.getGeneratorAtTimestamp(apiContext, genesisTimestamp - 100),
+			).rejects.toThrow('Invalid timestamp');
+		});
+	});
+
+	describe('getGeneratorsBetweenTimestamps', () => {
```

Comment:
this should test
- generators have more than 2 counts (ie: loop through the list)
- result is empty case(ie: timestamp is in sequential slot)
- not all generators are present in result